spec/models/pro_account_spec.rb

```diff
@@ -44,4 +44,43 @@
 
   end
 
+  describe '#active?' do
+
+    subject { FactoryGirl.create(:pro_account) }
+
+    let(:customer) do
+      Stripe::Customer.create(email: subject.user.email,
+                              source: stripe_helper.generate_card_token)
+    end
+
+    let(:plan) do
+      stripe_helper.create_plan(id: 'pro', amount: 1000)
+    end
+
+    before do
+      subject.update!(stripe_customer_id: customer.id)
+    end
+
+    it 'returns true if there is an active subscription' do
+      Stripe::Subscription.create(customer: customer,
+                                  plan: plan.id,
+                                  tax_percent: 20.0)
+      expect(subject.active?).to eq(true)
+    end
+
+    it 'returns true if there is an expiring subscription' do
+      subscription =
+        Stripe::Subscription.create(customer: customer,
+                                    plan: plan.id,
+                                    tax_percent: 20.0)
+      subscription.delete(at_period_end: true)
+      expect(subject.active?).to eq(true)
+    end
+
```

Comment:
I think an example of an instant cancellation (unlikely but we might end up banning someone) would be useful here - also demonstrates that someone with a fully cancelled subscription would be marked as not active